lib/spack/spack/build_systems/cuda.py

```diff
@@ -112,14 +116,14 @@ def cuda_flags(arch_list):
     conflicts('%clang@:3.7,4.1:',
               when='+cuda ^cuda@9.1' + arch_platform)
     conflicts('%clang@:3.7,5.1:', when='+cuda ^cuda@9.2' + arch_platform)
-    conflicts('%clang@:3.7,6.1:', when='+cuda ^cuda@10.0.130' + arch_platform)
-    conflicts('%clang@:3.7,7.1:', when='+cuda ^cuda@10.1.105' + arch_platform)
+    conflicts('%clang@:3.7,6.1:', when='+cuda ~allow-unsupported-compilers ^cuda@10.0.130' + arch_platform)
```

Comment:
Should we just if block this all out? Since it's not really allowing unsupported compilers?

Reply:
I don't think that's possible, the spec isn't available at the module level.